src/apps/frontend/components/otp/index.tsx

```diff
@@ -0,0 +1,81 @@
+import React, {
+  useRef,
+  useState,
+  FocusEventHandler,
+} from 'react';
+
+import constant from '../../constants';
+import { AsyncError, KeyboardKeys } from '../../types';
+
+import OTPInput from './otp-input';
+import Flex from '../flex/flex.component';
+import FlexItem from '../flex/flex-item.component';
+
+interface OTPProps {
+  error: string;
+  isLoading: boolean;
+  onBlur?: FocusEventHandler<HTMLInputElement>;
+  onChange: (values: string[]) => void;
+  onError: (error: AsyncError) => void;
+}
+
+const OTP: React.FC<OTPProps> = ({
+  error, isLoading, onBlur, onChange,
+}) => {
+  const [otp, setOtp] = useState<string[]>(Array(constant.OTP_LENGTH).fill(''));
+
+  const inputRef = useRef<HTMLInputElement[]>([]);
+
+  const handleOTPChange = (
+    inputValue: string,
+    index: number,
+  ): void => {
+    const otpInputs = [...otp];
+
+    if (inputValue.length >= 2) return;
```

Comment:
What does 2 represent? What is this check trying to do?

Reply:
Does maxlength =1 on input does not work?

Reply:
Do we know why? Is this is an issue with react or simple html? I looked at online and tried briefly and seemed to work